compatibility-tests/compile-fail/tests/ui/attribute-duplication.stderr

```diff
@@ -1,95 +1,107 @@
-error: Multiple 'visibility' attributes are not supported on an error enum; found a duplicate here:
- --> $DIR/attribute-duplication.rs:7:10
+error: Multiple 'visibility' attributes are not supported on an error enum
+ --> $DIR/attribute-duplication.rs:6:12
   |
-7 |     enum EnumError {
-  |          ^^^^^^^^^
+6 |     #[snafu(visibility(pub))]
+  |            ^^^^^^^^^^^^^^^^^
 
-error: Multiple 'source' attributes are not supported on one field; found a duplicate here:
-  --> $DIR/attribute-duplication.rs:25:13
+error: Multiple 'source' attributes are not supported on one field
+  --> $DIR/attribute-duplication.rs:23:20
    |
-25 |             source4: String,
-   |             ^^^^^^^
+23 |             #[snafu(source)]
+   |                    ^^^^^^^^
 
-error: Multiple 'backtrace' attributes are not supported on one field; found a duplicate here:
-  --> $DIR/attribute-duplication.rs:34:13
+error: Multiple 'backtrace' attributes are not supported on one field
+  --> $DIR/attribute-duplication.rs:25:20
    |
-34 |             backtrace3: String,
-   |             ^^^^^^^^^^
+25 |             #[snafu(backtrace(delegate))]
+   |                    ^^^^^^^^^^^^^^^^^^^^^
 
-error: Multiple 'source' attributes are not supported within an error variant; found a duplicate here:
-  --> $DIR/attribute-duplication.rs:18:13
+error: Multiple 'backtrace' attributes are not supported on one field
+  --> $DIR/attribute-duplication.rs:33:20
    |
-18 |             source2: Box<EnumError>,
-   |             ^^^^^^^
+33 |             #[snafu(backtrace)]
+   |                    ^^^^^^^^^^^
 
-error: Multiple 'source' attributes are not supported within an error variant; found a duplicate here:
-  --> $DIR/attribute-duplication.rs:21:13
+error: Multiple 'backtrace' attributes are not supported on one field
+  --> $DIR/attribute-duplication.rs:34:20
    |
-21 |             source3: String,
-   |             ^^^^^^^
+34 |             #[snafu(backtrace)]
+   |                    ^^^^^^^^^^^
 
-error: Multiple 'source' attributes are not supported within an error variant; found a duplicate here:
-  --> $DIR/attribute-duplication.rs:25:13
+error: Multiple 'source' attributes are not supported within an error variant
+  --> $DIR/attribute-duplication.rs:16:20
    |
-25 |             source4: String,
-   |             ^^^^^^^
+16 |             #[snafu(source(from(EnumError, Box::new)))]
```

Comment:
The order comes the fact that I'm extending the overall error list with the errors from each `SyntaxErrors`/`AtMostOne` as I `finish` it.  I think the way to achieve overall ordering would be to store the line number with each element as we add it, and then sort the final error list by that number.  The line number can be retrieved through `Spanned` on the attribute, then `.start` to get `LineColumn`.`line`.  Do you think that's worthwhile?

Reply:
I don't know that it's *super* worth it, mostly because the error test is a pathological case — I certainly hope that people aren't adding that many wrong / duplicate attributes at once! 

The main benefit I see is that our test output would be more stable. Maybe just a small thing to do in the future, if it becomes annoying.

Reply:
> our test output would be more stable

I'm curious what you mean on this one - I'd say the errors are "stable" in that they're always generated in the same order, and so can be consistently tested.  They're not ordered by line number, something that would be helpful for humans looking at a full error listing.  (And I agree, hopefully people aren't adding so many errors that this would be a problem...)

Reply:
I mostly mean that the order of things in this standard error file are based on the order of the code. If we do some "trivial" refactoring and flip the order of two calls, then this file will be invalidated, even though the errors are the same.

I also know that as this file grows, we will get a lot of incidental diffs; line numbers changing because we added/removed a line at the top. If that proves to be a problem, I might split it out into multiple files just to reduce that kind of churn.

Reply:
Ahh, I understand.  The big compile-fail tests for this bugged me anyway, so I'll split those out, and hopefully that will address the ordering fairly well too.

Reply:
To clarify, I mean that the order of the code *in snafu-derive* will affect the order of the generated errors.

You don't have to split out the compile-fail tests into multiple right now if it's not relevant; I mostly gave that as another example of the kind of "instability" of these tests

---

Truth be told, I'm not a big fan of this style of test, where the entire snapshot is captured. I find them overly brittle in general, but I wanted to use the better-supported trybuild library, which only has snapshots.

Reply:
> To clarify, I mean that the order of the code in snafu-derive will affect the order of the generated errors.

Yep, I realized that part after your hint in the last message.  I split out the tests so that there aren't multiple unrelated error checks being tested in one file, which means that reordering the code in snafu-derive won't unnecessarily break one of the compile-fail tests.  I also think it made it easier to understand which error case was being checked, when there aren't a bunch of other attributes/variants in the same test case.  (Test cases can also be split into modules for clarity, but that doesn't help with the irrelevant changes to line numbers in the snapshots.)